### PR TITLE
add maxBytes property to LabelInput

### PR DESCRIPTION
**Pull request by Bettelstab**

### CODE REVIEW

**Comment by Bettelstab** [comment]
Checking some things before reopening...

File: src/components/LabelInput.vue [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
         }
+        
+        public input() {
+            if (this.maxBytes !== undefined) {
+                const lengthInBytes = Utf8Tools.stringToUtf8ByteArray(this.liveValue!)
+                                        .byteLength;
+                if (lengthInBytes > this.maxBytes) {
+                   this.liveValue = this.lastValue;
+                   return;
+                }
```
Comment (sisou) [inline src/components/LabelInput.vue]: Looks like the indentation is off in this `if` block.

File: src/components/LabelInput.vue [hunk 2]
```diff
@@ -1,30 +1,47 @@
...
+        private lastEmittedValue = this.value;
 
         public focus() {
             (this.$refs.input as HTMLInputElement).focus();
         }
+        
+        public input() {
+            if (this.maxBytes !== undefined) {
+                const lengthInBytes = Utf8Tools.stringToUtf8ByteArray(this.liveValue!)
+                                        .byteLength;
```
Comment (sisou) [inline src/components/LabelInput.vue]: Can this be one line?

File: src/components/LabelInput.vue [hunk 3]
```diff
@@ -1,30 +1,47 @@
...
 
         private liveValue = this.value;
         private lastValue = this.value;
+        private lastEmittedValue = this.value;
 
         public focus() {
             (this.$refs.input as HTMLInputElement).focus();
         }
+        
+        public input() {
```
Comment (sisou) [inline src/components/LabelInput.vue]:
```suggestion
        private onInput() {
```

**Comment by Bettelstab** [comment]
@sisou Please check again :)

**Comment by danimoh** [comment]
Note that many of these changes are already contained in `soeren/new-designs`.
However, in that branch, the class was renamed to `Input`.
Therefore, spend special attention to these classes, when merging `soeren/new-designs` at some point.

**Pull request merged**